src/Ros2ForUnity/Scripts/Time/ROS2Clock.cs

```diff
@@ -19,57 +19,46 @@ namespace ROS2
 {
 
 /// <summary>
-/// A ros2 clock wrapper class that can acquire ros2 time on demand. It does not include a publisher.
-/// (Not every clock needs to be associated with a publisher)
+/// A ros2 clock class that for interfacing between a time source (unity or ros2 system time) and ros2cs messages, structs. 
 /// </summary>
 public class ROS2Clock
 {
-    private int seconds;
-    private uint nanoseconds;
-    private ROS2.Clock clock;
+    private ITimeSource _timeSource;
 
-    public ROS2Clock()
-    {
-        clock = new ROS2.Clock();
+    public ROS2Clock() : this(new DotnetTimeSource())
+    {   // By default, use DotnetTimeSource
```

Comment:
Why though? Shouldn't ros2 be the default?

Reply:
I also changed this to ros2 after some consideration